server/etcdserver/api/membership/cluster.go

```diff
@@ -83,14 +83,16 @@ const (
 func NewClusterFromURLsMap(lg *zap.Logger, token string, urlsmap types.URLsMap, opts ...ClusterOption) (*RaftCluster, error) {
 	c := NewCluster(lg, opts...)
 	for name, urls := range urlsmap {
-		m := NewMember(name, urls, token, nil)
-		if _, ok := c.members[m.ID]; ok {
-			return nil, fmt.Errorf("member exists with identical ID %v", m)
-		}
-		if uint64(m.ID) == raft.None {
-			return nil, fmt.Errorf("cannot use %x as member id", raft.None)
+		for idx := range urls {
+			m := NewMember(name, urls[idx:idx+1], token, nil)
+			if _, ok := c.members[m.ID]; ok {
+				return nil, fmt.Errorf("member exists with identical ID %v", m)
```

Comment:
You might be heading in the wrong direction. Usually each member has different peerURL(s), they must have different memberID. FYI. https://github.com/etcd-io/etcd/blob/f215cd89d2e2d24def615bb6da4d0f3038ddc3f9/server/etcdserver/api/membership/member.go#L63-L76

For simplicity, we can just add a warning message on startup if `--name `isn't provided a value.

Reply:
Thank you very much for the reminder.